FRPs/proposal-27-alternative-architectures.md

```diff
@@ -0,0 +1,32 @@
+---
+id: 27
+title: Architecture on Other Chains
+team: Sikka
+created: 2020-12-13
+paper: 1
+---
+
+# Flashbots on Alternative Chain Architectures
+
+## Background and Problem Statement
+The Layer 1 MEV Auction design presented in Paper 1 (aka *Flashbots*) is particularly architected for use in Proof of Work Nakamoto consensus based blockchains (more particularly Ethereum).  However, as alternative blockchain architectures become more and more prevalent due to rise of "third gen blockchains" (including Eth 2.0, Cosmos, Avalanche, Solana, etc), it is of interest to understand what chain architectures *Flashbots* can be easily ported to.  For example, can *Flashbots* be used on Ethereum as it shifts to Eth 2.0?  How about Tendermint BFT-based chains?  How about leaderless consensus protocols like Avalanche?  Furthermore, are there any differences in the impact of high REV on these different chains vs Ethereum 1.0?  
+
+By understanding what chains *Flashbots* (and other Layer 1 MEV Auction architectures) can be used to efficiently extract MEV, as well as the properties these chains share, we can better understand how to mitigate those properties in novel architectures to help minimize MEV in next-generation blockchains.
+
+## Plan and Deliverables
+- Identify what types of MEV-extracting "actions" *Flashbots* helps enable
+- Describe properties required of a blockchain architecture for above MEV-extracting "actions" to be executable
+- Describe properties required of a blockchain architecture to allow *Flashbots*'s Layer 1 MEV Auction architecture to function
+- Analyze alternative consensus systems of interest (with no state machine changes) to see which of the above properties they satisfy to determine if the Flashbots architecture would be compatible. Particularly:
+  - Nakamoto-style Proof of Stake (Eth 2.0 Casper)
+  - Classical BFT (Tendermint/Cosmos)
+  - Leaderless Consensus (Avalanche) (MAYBE)
+- Analyze the above consensus protocols to see what the impacts of high extracted MEV is and how it differs from Nakamoto Proof of Work (paricularly with respect to consensus instability).
```

Comment:
Does the consensus type really matter? I can see Nakamoto ones being more vulnerable to Time Bandit attacks due to the nature of reorgs. BFT ones do not seem vulnerable to time bandits in the single chain context, but might experience performance degradation during the precommit/commit phase.

I think it's interesting to see how time bandit attacks would look like in the Cosmos/IBC cross-chain context.

I can also see MEV-combating techniques which separate ordering from execution reducing MEV to 0 in all contexts. In your talk you say that you don't like "Signed Receipts", but I did not hear a convincing argument against them, could you elaborate on that perhaps? Agree re: MEV auctions not solving the problem. As for Batch Auctions, you may want to generalize this to any sort of batching. You could very easily argue that L2 Plasma/Rollup systems reduce MEV by reducing the number of txpool permutations available to a miner

Reply:
> I can see Nakamoto ones being more vulnerable to Time Bandit attacks due to the nature of reorgs. BFT ones do not seem vulnerable to time bandits in the single chain context, but might experience performance degradation during the precommit/commit phase.

Yeah, I think these are both forms of "consensus instability".  The Time Bandit attacks in nakamoto chains create safety issues, but high MEV in BFT more likely causes liveness issues (the performance degradation might be permanent).  But I do think the impacts are way lower and more dealable-with in BFT chains.

> I think it's interesting to see how time bandit attacks would look like in the Cosmos/IBC cross-chain context.

So how IBC deals with Nakamoto chains, is that when opening the connection, it has to choose some threshold (how many blocks deep) at which it considers the block "pseudo-finalized".  I imagine the way IBC would deal with this is to look at data on REV and past time bandit attacks on the chain its connecting to in order to make more informed decisions on what pseudo-finality threshold to use.  (On a side note, when it comes to MEV-combating, I think there's some interesting stuff to be done in cross-chain environment related to nested threshold decryption).

> I can also see MEV-combating techniques which separate ordering from execution reducing MEV to 0 in all contexts.

What do you mean by separating ordering from execution?  Execution is always done by everyone.  Do you mean separating inclusion from ordering?

> In your talk you say that you don't like "Signed Receipts", but I did not hear a convincing argument against them, could you elaborate on that perhaps?

Signed receipts completely depends on the honesty of the proposer who's creating the signed receipts.  They can easily choose to censor your txs or insert their own before giving you a signed receipt.  Also, without any sort of encryption of txs, the proposer who is creating the signed receipts can easily to Dark Forest attacks.  

> As for Batch Auctions, you may want to generalize this to any sort of batching.

Yeah, in the talk I refer to it as "Batch Execution" in order to specify that it be any sort of batching system that executes in the way that reduces importance of ordering in a block.

? You could very easily argue that L2 Plasma/Rollup systems reduce MEV by reducing the number of txpool permutations available to a miner

Are you referring to using MEV auctions to remove ordering powers from the operator?  This shifts the ordering power away from the operator to a third party, but the ideal would be to give the ordering power to a source of fair randomness, which will be the most neutral.

Reply:
One potential thing to consider: How much do liveness faults reorder transactions within a block? One thing I remember in Stellar's codebase is that there is a weird griefing attack of the following form:

1. Validator proposes block `B = [tx1|tx2|...|txn]`
2. Attacker DDoS's >2/3 of validators, causes a liveness fault / view change
3. Upon view change, a _new_ validator is the proposer and proposes a block with the same transactions in a different order, e.g. `B' = [txn|tx2|...|tx1]`
4. Attacker stops DDoSing when the block has the order they want

Often times, this attack works because non-proposing validators will vote / sign on a hash of the _set_ of transactions (inclusion) not the ordering.